adr/2022_03_24_remove_flush_from_handlers.md

```diff
@@ -0,0 +1,25 @@
+# Remove flush from handlers
+
+* Status: accepted
+* Date: 2022-03-24
+
+## Context and Problem Statement
+
+Using flush() in command handlers can cause problems with inconsistent DB state. Doctrine always uses transactions but on
+flush() Doctrine will execute all the changes that have been tracked so far. Implicitly flushing will commit all the 
+changes queued up so far. Not just the entity you just persisted. This is also one of the reasons why it is dangerous to
+just call flush.
+
+## Decision Drivers
+
+* avoid inconsistent data in DB
+* flexibility to rollback changes
+
+## Decision Outcome
```

Comment:
```suggestion

## Considered Options

### Flushing in handlers 

Using flush() in command handlers can cause problems with inconsistent DB states if the end-user decides to adjust models' definitions. Implicitly flushing will commit all the changes queued up so far, which may lead to DB error or committing not consistent data. What is more, we would need an additional feature that would allow adjusting the entity's state before the transaction will close.

### Flushing outside of handlers with the manually triggered transaction

Manual setting up of transactions around handlers clearly defines their boundaries. Even tho this may still lead to DB error or committing not consistent data, setting boundaries outside of our code improves DX greatly, as an adjustment on objects may be made with decorator pattern outside of predefined handler.

## Decision Outcome
```

Probably you should also extract some good and bad points from my descriptions.